web_programming/instagram_pic.py

```diff
@@ -0,0 +1,16 @@
+import requests
+from bs4 import BeautifulSoup
+import datetime
+
+if __name__ == "__main__":
+    url = input("Enter image url: ")
+    print("Downloading image...")
+    req = requests.get(url)
+    soup = BeautifulSoup(req.content, "html.parser")
+    metaTag = soup.find_all("meta", {"property": "og:image"}) #get all meta tags
+    imgURL = metaTag[0]["content"]
+    r = requests.get(imgURL)
+    fileName = datetime.datetime.now().strftime("%Y-%m-%d_%H:%M:%S") + ".jpg"
+    with open(fileName, "wb") as fp:
+        fp.write(r.content)
```

Comment:
```suggestion
    image_data = requests.get(imgURL).content
    file_name = f"{datetime.now():%Y-%m-%d_%H:%M:%S}.jpg"
    with open(file_name, "wb") as fp:
        fp.write(image_data)
```
Avoid single-letter variable names -- They make code look like it was written in the 1970's.  The reader of this code does not care about the response, but they do care about the image_data so focus their attention on that.

Use f-strings which are more expressive especially with complex types like datetimes.